internal/globallogger/globallogger.go

```diff
@@ -47,6 +48,21 @@ func IsInitialized() bool {
 	return globalLogger != nil
 }
 
+// forceSyncer implements the zapcore.CheckWriteHook interface and ensures that sync is called on the provided core.
+// As per the advice from https://pkg.go.dev/go.uber.org/zap#WithFatalHook, os.Exit(1) is called to halt execution after
+// Sync has completed
+type forceSyncer struct {
+	core zapcore.Core
+}
+
+var _ zapcore.CheckWriteHook = &forceSyncer{}
+
+// OnWrite calls sync on the underlying core and then calls os.Exit(1)
+func (f *forceSyncer) OnWrite(_ *zapcore.CheckedEntry, _ []zapcore.Field) {
+	f.core.Sync()
```

Comment:
nit: what happens to an error in there? Could we at least make it explicit that we're ignoring it?

Reply:
Updated!